server/routes.go

```diff
@@ -197,20 +202,32 @@ func GenerateHandler(c *gin.Context) {
 
 	checkpointLoaded := time.Now()
 
-	prompt := req.Prompt
-	if !req.Raw {
-		prompt, err = model.Prompt(req)
+	var prompt string
+	sendContext := false
+	switch {
+	case req.Raw:
+		prompt = req.Prompt
+	case areMessagesSet:
```

Comment:
What happens if `len(req.Messages) == 0`? That should probably just load the model like an empty prompt